extensions-core/s3-extensions/src/main/java/org/apache/druid/storage/s3/S3TaskLogs.java

```diff
@@ -146,6 +147,17 @@ private void pushTaskFile(final File logFile, String taskKey) throws IOException
     }
   }
 
+  private void uploadFileIfPossible(String bucket, String key, File file)
+  {
+    final PutObjectRequest logFilePutRequest = new PutObjectRequest(bucket, key, file);
+
+    if (!config.getDisableAcl()) {
+      logFilePutRequest.setAccessControlList(S3Utils.grantFullControlToBucketOwner(service, bucket));
+    }
+    log.info("Pushing [%s] to bucket[%s] and key[%s].", file, bucket, key);
+    service.putObject(logFilePutRequest);
+  }
+
```

Comment:
@dclim, uploadFileIfPossible(..) is now a common method with my change in S3DataSegmentPusher.java and S3TaskLogs.java. Should I extract this method and move it to S3Utils.java. The only worry is the log statement in the method will now be moved to utils class. Is that ok ?

Reply:
Yes that's a good idea and more maintainable than duplicating the code in `S3TaskLogs`. It's not a problem that the logger class changes.

Reply:
Moved this method to S3Utils.java.